services/api/src/resources/billing/resolvers.ts

```diff
@@ -0,0 +1,136 @@
+export const addBillingModifier: AddBillingModifierAlias = async (root, args, context) => {
+  const { hasPermission } = context;
+  const { input } = args;
+
+  // Input Validation
+  if (R.isEmpty(input.group)) {
+    throw new Error('You must provide a billing group name or id');
+  }
+
+  if (R.isEmpty(R.omit(['group', 'startDate', 'endDate'], input))) {
+    throw new Error('You must provide a discount value or extra cost.');
+  }
+
+  if( new Date(input.startDate) >= new Date(input.endDate)) {
+    throw new Error('You must provide a start date before the end date.');
+  }
+
+  // Permissions
+  await hasPermission('group', 'add');
+
+  // Action
+  return billingModel.addBillingModifier(input);
+};
+
+/**
+ * Update Billing Modifier
+ *
+ * @param {obj} root The rootValue passed from the Apollo server configuration.
+ * @param {UpdateBillingModifierInput} args
+ * @param {ExpressContext} context this includes the context passed from the apolloServer query
+ *     { sqlClient, hasPermissions, keycloakGrant, requestCache }
+ *
+ * @return {String} Success
+ */
+export const updateBillingModifier = async (_, args: UpdateBillingModifierInput, context: {models: any, hasPermission: any}) => {
+  const { hasPermission } = context;
+  const { input : { id, patch } } = args;
+
+  // Permissions
+  await hasPermission('group', 'update');
+
+  // Action
+  return billingModel.updateBillingModifier(id, patch);
+}
+
+/**
+ * Delete Billing Modifier
+ *
+ * @param {obj} root The rootValue passed from the Apollo server configuration.
+ * @param {DeleteBillingModifierInput} args
+ * @param {ExpressContext} context this includes the context passed from the apolloServer query
+ *     { sqlClient, hasPermissions, keycloakGrant, requestCache }
+ *
+ * @return {String} Success
+ */
+export const deleteBillingModifier = async (
+  _,
+  args: DeleteBillingModifierInput,
+  context: {models: any,
+  hasPermission: any}
+) => {
+  const { hasPermission } = context;
+  const { input: { id } } = args;
+
+  // Permissions
+  await hasPermission('group', 'delete');
+
+  // Action
+  return billingModel.deleteBillingModifier(id);
+}
+
+/**
+ * Delete All Billing Modifiers for a given Billing Gropu
+ *
+ * @param {obj} root The rootValue passed from the Apollo server configuration.
+ * @param {DeleteAllBillingGroupModifiersInput} args
+ * @param {ExpressContext} context this includes the context passed from the apolloServer query
+ *     { sqlClient, hasPermissions, keycloakGrant, requestCache }
+ *
+ * @return {String} Success
+ */
+export const deleteAllBillingModifiersByBillingGroup = async (
+  _,
+  args: DeleteAllBillingGroupModifiersInput,
+  context: {models: any,
+  hasPermission: any}
+) => {
+  const { hasPermission } = context;
+  const { input } = args;
+
+  // Permissions
+  await hasPermission('group', 'delete');
```

Comment:
There is a specific "deleteAll" scope that is currently granted only to Admins. The main reason we have deleteAll* api mutations is to support the `local-api-data-watcher-pusher` service. These generally aren't ever used on a regular basis.

So... do you think it's useful to let "normal" users delete all billing modifiers for a group (in which case, it needs the group id attribute)? The alternative is to change this mutation to delete all modifiers (for all groups) and switch to `deleteAll` scope.

Reply:
This was definitely added to help me with tests with a very slight potential that this could be useful... Not sure it's needed though

Reply:
Addressed in bcb60674748dc28ef2fa5f75027f9c618e78950c but let's talk more about permissions.